cdap-ui/app/features/hydratorplusplus/controllers/create/partials/postactions-ctrl.js

```diff
@@ -0,0 +1,191 @@
+      name: config.plugin.name,
+      type: config.plugin.type
+    };
+
+    this.pluginConfig = config;
+    this.pluginFetch(tab);
+
+    this.activeTab = tab;
+  }
+
+  // Fetching Backend Properties
+  pluginFetch(tab) {
+    let params = {
+      namespace: this.$state.params.namespace,
+      pipelineType: tab.artifact.name,
+      version: tab.artifact.version,
+      scope: tab.artifact.scope,
+      extensionType: tab.type,
+      pluginName: tab.name
+    };
+
+    this.myPipelineApi.fetchPluginProperties(params)
+      .$promise
+      .then( (res) => {
+        this.pluginConfig._backendProperties = res[0].properties;
+        this.fetchWidgets(tab);
+      });
+  }
+
+  // Fetching Widget JSON for the plugin
+  fetchWidgets(tab) {
+    let artifact = {
+      name: tab.artifact.name,
+      version: tab.artifact.version,
+      scope: tab.artifact.scope,
+      key: 'widgets.' + tab.name + '-' + tab.type
+    };
+    this.HydratorPlusPlusPluginConfigFactory
+      .fetchWidgetJson(artifact.name, artifact.version, artifact.scope, artifact.key)
+      .then( (widgetJson) => {
+        this.noConfig = false;
+
+        this.groupsConfig = this.HydratorPlusPlusPluginConfigFactory.generateNodeConfig(this.pluginConfig._backendProperties, widgetJson);
+
+        // Initializing default value
+        angular.forEach(this.groupsConfig.groups, (group) => {
+          angular.forEach(group.fields, (field) => {
+            if (field.defaultValue) {
+              this.pluginConfig.properties[field.name] = this.pluginConfig.properties[field.name] || field.defaultValue;
+            }
+          });
+        });
+      }, () => {
+        this.noConfig = true;
+      });
+  }
+
+  cancel() {
+    this.activeTab = 'OVERVIEW';
+    this.pluginConfig = {};
+  }
+
+  fieldValidation() {
+    let isValid = true;
+    angular.forEach(this.pluginConfig._backendProperties, (value, key) => {
+      if (value.required && !this.pluginConfig.plugin.properties[key]) {
+        isValid = false;
+      }
+    });
+
+    return isValid;
+  }
+
+  save(isEdit) {
+    let isValid = this.fieldValidation();
+
+    if (!isValid) {
+      this.myAlertOnValium.show({
+        type: 'danger',
+        content: 'Please fill out all required field'
+      });
+      return;
+    }
+
+    delete this.pluginConfig._backendProperties;
+
+    if (isEdit) {
+      this.HydratorPlusPlusConfigActions.editPostAction(angular.copy(this.pluginConfig));
+
+      this.myAlertOnValium.show({
+        type: 'success',
+        content: this.pluginConfig.plugin.name + ' post action saved.'
+      });
+    } else {
+      this.HydratorPlusPlusConfigActions.addPostAction(angular.copy(this.pluginConfig));
+
+      this.myAlertOnValium.show({
+        type: 'success',
+        content: this.pluginConfig.plugin.name + ' post action added.'
+      });
+    }
+
+    this.pluginConfig = {};
+  }
+
+  deleteAction(config) {
+    this.HydratorPlusPlusConfigActions.deletePostAction(config);
```

Comment:
This should highlight fields in the form (just like while saving/adding), just like we do in node configuration.